File: CS/2130_Maximum_Twin_Sum_of_a_Linked_List.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public int PairSum(ListNode head) {
        List<int> list = new List<int>();
        while(head != null){
            list.Add(head.val);
            head = head.next;
        }
        int result = 0;
        int h = list.Count;
        for(int i = 0; i < h/2; i++){
            result = Math.Max(result, list[i] + list[h-i-1]);
        }
        return result;
    }
}

File: Java/2130_Maximum_Twin_Sum_of_a_Linked_List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    private List<Integer> list = new ArrayList<>();
    
    public int pairSum(ListNode head) {
        while(head != null){
            list.add(head.val);
            head = head.next;
        }
        int result = 0;
        for(int i = 0; i < list.size(); i++){
            result = Math.max(result, list.get(i) + list.get(list.size()-1 - i));
        }
        return result;
    }
}
/*
5 4 2 1 0 3
      s
            f
    t
2 4 5
p
t

null
p

ListNode slow = head, fast = head, prev = null;
     
        while (fast != null && fast.next != null) {
            ListNode temp = slow;
            slow = temp.next;
            fast = fast.next.next;
            temp.next = prev;
            prev = temp;
        }
        
        int max = 0;
        while (slow != null) {
            max = Math.max(max, prev.val+slow.val);
            slow = slow.next;
            prev = prev.next;
        }
        return max;
    }

*/


